lib/js/components/Form/SelectionControl/SelectionControl.vue

```diff
@@ -0,0 +1,352 @@
+		class="selectionControl"
+		:class="[
+			`-${size}`,
+			{
+				'-selected': isSelected,
+				'-default': state === SELECTION_CONTROL_STATE.DEFAULT,
+				'-disabled': state === SELECTION_CONTROL_STATE.DISABLED,
+				'-loading': state === SELECTION_CONTROL_STATE.LOADING,
+				'-focused': isFocused,
+			},
+		]"
+		@click="onToggle"
+	>
+		<div class="selectionControl__iconWrapper">
+			<icon :icon="icon" :size="iconSize" class="selectionControl__icon" />
+		</div>
+		<div v-if="label" class="selectionControl__labelWrapper">
+			<div class="selectionControl__label">{{ label }}</div>
+		</div>
+		<input
+			type="checkbox"
+			class="selectionControl__checkbox"
+			:value="isSelected"
+			@change="onToggle"
+			@focus="onFocus"
+			@blur="onBlur"
+		/>
+	</div>
+</template>
+
+<style scoped lang="scss">
+@import '../../../../styles/settings/animations';
+@import '../../../../styles/settings/colors/tokens';
+@import '../../../../styles/settings/spacings';
+@import '../../../../styles/settings/typography/tokens';
+
+$selection-control-sizes: (
+	'x-small': (
+		'gap': $space-xxxxs,
+		'iconWrapperPadding': $space-xxxs,
+		'labelWrapperPadding': $space-xxxxs,
+		'iconBorder': $space-xxxs,
+	),
+	'small': (
+		'gap': $space-xxs,
+		'iconWrapperPadding': $space-xxxs,
+		'labelWrapperPadding': $space-xxxs,
+		'iconBorder': $space-xxxs,
+	),
+	'medium': (
+		'gap': $space-xxs,
+		'iconWrapperPadding': $space-xxs,
+		'labelWrapperPadding': $space-xxs,
+		'iconBorder': $space-xxs,
+	),
+);
+
+$selection-control-colors: (
+	'default': (
+		'not-selected': (
+			'label': $color-neutral-text-heavy,
+			'icon': $color-neutral-icon,
+		),
+		'selected': (
+			'label': $color-neutral-text-heavy,
+			'icon': $color-primary-icon,
+		),
+		'hoverable': true,
+	),
+	'disabled': (
+		'not-selected': (
+			'label': $color-neutral-text-heavy-disabled,
+			'icon': $color-neutral-icon-disabled,
+		),
+		'selected': (
+			'label': $color-neutral-text-heavy-disabled,
+			'icon': $color-primary-icon-disabled,
+		),
+	),
+	'loading': (
+		'not-selected': (
+			'label': $color-neutral-text-heavy,
+			'icon': $color-neutral-icon,
+		),
+		'selected': (
+			'label': $color-neutral-text-heavy,
+			'icon': $color-primary-icon,
+		),
+	),
+);
+
+.selectionControl {
+	$root: &;
+
+	@each $name, $map in $selection-control-colors {
+		@if map-get($map, 'hoverable') {
+			&.-#{$name} {
+				@each $size, $map in $selection-control-sizes {
+					&.-#{$size} {
+						#{$root}__icon::before {
+							bottom: -#{map-get($map, 'iconBorder')};
+							left: -#{map-get($map, 'iconBorder')};
+							right: -#{map-get($map, 'iconBorder')};
+							top: -#{map-get($map, 'iconBorder')};
+						}
+					}
+				}
+
+				cursor: pointer;
+
+				#{$root}__icon::before {
+					background-color: transparent;
+					border-radius: 50%;
+					content: '';
+					position: absolute;
+					transform: scale(0);
+					transition: all $default-transition-time ease 0ms;
+					z-index: -1;
+				}
+
+				&:hover,
```

Comment:
I'm not convinced that defining these styles in for loop makes it easier to understand. What would you say about moving it into regular styles block